lib/bolt/transport/ssh/connection.rb

```diff
@@ -130,7 +130,11 @@ def connect
 
         def disconnect
           if @session && !@session.closed?
-            @session.close
+            begin
+              Timeout.timeout(@target['connect-timeout']) { @session.close }
```

Comment:
Re-examining this work shows that I had introduced a (pretty serious) bug. I should have used `@target.options['connect-timeout']` by omitting  `.options` a `NoMethodError` was raised here as a result of trying to use a `[]` accessor on a `Target` object. This was caught in the blanket `StandardError` handling at https://github.com/puppetlabs/bolt/blob/d739a203bc4a7e5afc4c2b650517e1c50fa153ab/lib/bolt/transport/ssh.rb#L74 . So I think this resulted in the `shutdown!` method never being called.